src/main/java/net/minecraftforge/fml/common/FMLModContainer.java

```diff
@@ -595,16 +597,34 @@ public void constructMod(FMLConstructionEvent event)
     @Subscribe
     public void handleModStateEvent(FMLEvent event)
     {
-        if (!eventMethods.containsKey(event.getClass()))
+        if (!eventHandlers.containsKey(event.getClass()))
         {
             return;
         }
         try
         {
-            for (Method m : eventMethods.get(event.getClass()))
-            {
-                m.invoke(modInstance, event);
-            }
+            eventHandlers.get(event.getClass())
+                    .stream()
+                    .map(eventHandler ->
+                    {
+                        // safely transform the result of eventHandler.handleEvent into a CompletableFuture
+                        return new CompletableFuture<Void>().thenCompose(aVoid ->
```

Comment:
This new `CompletableFuture` will never complete. Nobody ever calls `complete` or `completeExceptionally` on it.

Reply:
@diesieben07 right, I confused the behaviour of CompletableFuture.completedFuture with that of the CompletableFuture constructor.